python_modules/libraries/dagster-components/dagster_components/core/component.py

```diff
@@ -94,13 +107,45 @@ def get_entry_points_from_python_environment(group: str) -> Sequence[importlib.m
 
 
 COMPONENTS_ENTRY_POINT_GROUP = "dagster.components"
+BUILTIN_PUBLISHED_COMPONENT_ENTRY_POINT = "dagster_components"
+BUILTIN_TEST_COMPONENT_ENTRY_POINT = "dagster_components.test"
 
 
 class ComponentRegistry:
     @classmethod
-    def from_entry_point_discovery(cls) -> "ComponentRegistry":
+    def from_entry_point_discovery(
+        cls, use_test_component_lib: bool = False
+    ) -> "ComponentRegistry":
+        """Discover components registered in the Python environment via the `dagster_components` entry point group.
+
+        `dagster-components` itself registers two component entry points: `dagster_components` and
+        `dagster_components.test`. The `dagster_components` entry point resolves to published
+        our components. The `dagster_components.test` entry point resolves to a separate set of test
```

Comment:
Hmmm why not make this more pluggable? Won't we potentially want multiple test libraries? How about just make it parameterizable?